### DISCUSSION TITLE
"Path" argument must be of type string

### DISCUSSION THREAD

**Discussion by DeepSky8** [Phoscur/oryx-macro-hax#9] [Q&A] [answered]
**First off, THANK YOU for your work on this tool. It's a huge help to me.**

Per your recommendation, I updated to the latest version. I'm still receiving the following error when running `npm run get -- [hash]`
                                                                                

> Downloaded layout "0LXoN" to ./layout_src (from https://oryx.zsa.io/rails/active
> _storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNTAxUUE9PSIsImV4cCI6
> bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bb15af0be4346b3ef05dc6d4b34036ec6d80b9ad/moonl
> ander_qwerty-js_0LXoN_Jyz49.zip).                                               
> TypeError: The "path" argument must be of type string. Received undefined       
>     at new NodeError (node:internal/errors:400:5)                               
>     at validateString (node:internal/validators:163:11)                         
>     at join (node:path:429:7)                                                   
>     at C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\src\oryx-downloader.ts:53:29      

The keymap does download, so maybe it's fine. I'm just concerned when I see an error message.                                                                         

**Potential issues I've identified:**

- I'm not familiar with .env files; should it have a specific name?
- The Github Action Variables you link in the readme is not included in the build; is that the name I should use for the .env file?
- When running `npm start`, I receive the following error:

> Reading Keymap Source for User undefined: ./layout_src/undefined/keymap.c       
> C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\src\main.ts:16                    
>         throw new Error("USER_NAME cannot be empty");                           
>               ^                                                                 
> Error: USER_NAME cannot be empty                                                
>     at main (C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\src\main.ts:16:15)   
>     at Object.<anonymous> (C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\src\main.ts:33:5)                                                                      
>     at Module._compile (node:internal/modules/cjs/loader:1218:14)               
>     at Module.m._compile (C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\node_modules\ts-node\src\index.ts:439:23)                                               
>     at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)        
>     at Object.require.extensions.<computed> [as .ts] (C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\node_modules\ts-node\src\index.ts:442:12)                   
>     at Module.load (node:internal/modules/cjs/loader:1081:32)                   
>     at Function.Module._load (node:internal/modules/cjs/loader:922:12)          
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)                                                                       
>     at Object.<anonymous> (C:\Users\DeepS\qmk_firmware\ergodox-macro-hax\node_modules\ts-node\src\bin.ts:157:12)                                                

I reviewed my .env file and confirmed that I had entered my Github username in the USER_NAME field

Does anything jump out that I can do differently?

**Comment by Phoscur** [comment, marked as answer]
Oh, maybe I closed #7 too early? But you can still comment on it?

> The filename has to be `.env` not `process.env`!

Are you sure you are not mistakenly prefixing your environment file?

Please post the contents of your `.env` file, your node version & your platform too.

Edit: Added recently added USER_NAME config to README: https://github.com/Phoscur/oryx-macro-hax/commit/2ad3fdc552cd6d7d874707f7f5949d2cccadb638 (it's hard to miss in the `.env.dist` file though :P) 

Also shows your `.env` file is still considered missing and all the entries are `undefined`.

**Reply by DeepSky8** [reply]
You stated that as gently as possible; thank you. I was indeed prefixing the .env file. TIL they can stand on their own. That solved this issue. Many thanks!
